[ATfE] toolchain build fails on runtimes on macOS

**Summary**
Runtimes build fails when load LLVMExports.cmake

**To Reproduce**

cmake -G Ninja ..
ninja llvm-toolchain

**Expected behavior**
Build should complete.

**Actual behavior**
[14/1028] Performing compiler_rt-configure step for 'runtimes-aarch64a_be_soft_nofp_exn_rtti'
FAILED: /Users/azo/Code/arm-toolchain/arm-software/embedded/build/multilib-builds/runtimes/aarch64a_be_soft_nofp_exn_rtti/stamp/runtimes-aarch64a_be_soft_nofp_exn_rtti-compile
r_rt-configure
cd /Users/azo/Code/arm-toolchain/arm-software/embedded/build/multilib-builds/multilib/picolibc-build && /opt/homebrew/bin/cmake --build /Users/azo/Code/arm-toolchain/arm-softw
are/embedded/build/multilib-builds/runtimes/aarch64a_be_soft_nofp_exn_rtti/build --target compiler_rt-configure && /opt/homebrew/bin/cmake -E touch /Users/azo/Code/arm-toolcha
in/arm-software/embedded/build/multilib-builds/runtimes/aarch64a_be_soft_nofp_exn_rtti/stamp/runtimes-aarch64a_be_soft_nofp_exn_rtti-compiler_rt-configure
[0/1] Performing configure step for 'compiler_rt'
-- Could NOT find LibEdit (missing: LibEdit_INCLUDE_DIRS LibEdit_LIBRARIES)
CMake Error at /Users/azo/Code/arm-toolchain/arm-software/embedded/build/llvm/lib/cmake/llvm/LLVMExports.cmake:772 (add_library):
  ADD_LIBRARY called with SHARED option but the target platform does not
  support dynamic linking.
Call Stack (most recent call first):
  /Users/azo/Code/arm-toolchain/arm-software/embedded/build/llvm/lib/cmake/llvm/LLVMConfig.cmake:262 (include)
  cmake/Modules/CompilerRTUtils.cmake:312 (find_package)
  CMakeLists.txt:110 (load_llvm_config)
-- Configuring incomplete, errors occurred!

**Environment**
macOS Sequoia, homebrewed cmake, meson, ninja, python, qemu
both with AppleClang 17.0.0 and homebrewed clang 20.1.5

--- a/arm-software/embedded/arm-runtimes/CMakeLists.txt
+++ b/arm-software/embedded/arm-runtimes/CMakeLists.txt
@@ -253,14 +253,17 @@ endif()
 
 if(ENABLE_COMPILER_RT_TESTS)
     set(compiler_rt_test_flags "${lib_compile_flags} ${test_link_flags}")
-    set(compiler_rt_lit_args "${LLVM_LIT_ARGS} --xunit-xml-output=results.junit.xml")
+    set(compiler_rt_lit_args "${LLVM_LIT_ARGS} --path=${LLVM_BINARY_DIR}/bin --xunit-xml-output=results.junit.xml")
+    set(external_lit_path "${LLVM_BINARY_DIR}/bin/llvm-lit")
     set(
         compiler_rt_test_cmake_args
         -DCOMPILER_RT_INCLUDE_TESTS=ON
         -DCOMPILER_RT_EMULATOR=${lit_test_executor}
         -DCOMPILER_RT_TEST_COMPILER=${LLVM_BINARY_DIR}/bin/clang
         -DCOMPILER_RT_TEST_COMPILER_CFLAGS=${compiler_rt_test_flags}
+        -DLLVM_INCLUDE_TESTS=ON
         -DLLVM_LIT_ARGS=${compiler_rt_lit_args}
+        -DLLVM_EXTERNAL_LIT=${external_lit_path}
     )
 endif()
 
@@ -272,11 +275,11 @@ endif()
 
 ExternalProject_Add(
     compiler_rt
-    SOURCE_DIR ${llvmproject_src_dir}/compiler-rt
     STAMP_DIR ${PROJECT_PREFIX}/compiler_rt/${VARIANT_BUILD_ID}/stamp
     BINARY_DIR ${PROJECT_PREFIX}/compiler_rt/${VARIANT_BUILD_ID}/build
     DOWNLOAD_DIR ${PROJECT_PREFIX}/compiler_rt/${VARIANT_BUILD_ID}/dl
     TMP_DIR ${PROJECT_PREFIX}/compiler_rt/${VARIANT_BUILD_ID}/tmp
+    SOURCE_DIR ${llvmproject_src_dir}/runtimes
     INSTALL_DIR compiler-rt/install
     CMAKE_ARGS
     ${compiler_launcher_cmake_args}
@@ -303,7 +306,8 @@ ExternalProject_Add(
     -DCOMPILER_RT_BUILD_SANITIZERS=OFF
     -DCOMPILER_RT_BUILD_XRAY=OFF
     -DCOMPILER_RT_DEFAULT_TARGET_ONLY=ON
-    -DLLVM_CMAKE_DIR=${LLVM_BINARY_DIR}
+    -DLLVM_ENABLE_RUNTIMES=compiler-rt
+    -DRUNTIME_VARIANT_NAME=${VARIANT}
     -DLLVM_ENABLE_PER_TARGET_RUNTIME_DIR=ON
     ${compiler_rt_test_cmake_args}
     STEP_TARGETS configure build install

--- a/arm-software/embedded/arm-runtimes/test-support/modify-compiler-rt-xml.py
+++ b/arm-software/embedded/arm-runtimes/test-support/modify-compiler-rt-xml.py
@@ -29,7 +29,7 @@ def main():
     # Variants such as armv8m.main need to be renamed.
     variant_name = args.variant.replace(".", "_")
 
-    xml_file = os.path.join(args.dir, "test", "results.junit.xml")
+    xml_file = os.path.join(args.dir, "compiler-rt", "test", "results.junit.xml")
 
     tree = ElementTree.parse(xml_file)
     root = tree.getroot()

